### PR TITLE
Container kill

### PR DESCRIPTION
**Pull request by paigerube14**
### Description
Adding hard killing of containers in pods based on namespace, labels, or names of pods or container 

### Fixes
https://github.com/cloud-bulldozer/kraken/issues/60

### CODE REVIEW

File: docs/container_scenarios.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+####  Example Config
+The following are the components of Kubernetes/OpenShift for which a basic chaos scenario config exists today.
+
+```
+scenarios:
+- name: "<Name of scenario>"
+  namespace: "<specific namespace>" # can specify "*" if you want to find in all namespaces
+  label_selector: "<label of pod(s)>"
+  container_name: "<specific container name>"  # This is optional, can take out and will kill all pods in all pods found under namespace and label
```
Comment (mffiedler) [inline docs/container_scenarios.md:14]: should this be "and will kill all containers in all pods"?

Comment (paigerube14) [reply docs/container_scenarios.md:14]: Yes!

File: kraken/pod_scenarios/setup.py [hunk 2]
```diff
@@ -34,3 +37,89 @@ def run(kubeconfig_path, scenarios_list, config, failed_post_scenarios, wait_dur
     except Exception as e:
         logging.error("Failed to run scenario: %s. Encountered the following " "exception: %s" % (pod_scenario[0], e))
     return failed_post_scenarios
+
+
+def container_run(kubeconfig_path, scenarios_list, config, failed_post_scenarios, wait_duration):
+    logging.info("scenario lis t" + str(scenarios_list))
```
Comment (chaitanyaenr) [inline kraken/pod_scenarios/setup.py:43]: s/scenario lis t/scenario list

File: kraken/pod_scenarios/setup.py [hunk 3]
```diff
@@ -34,3 +37,89 @@ def run(kubeconfig_path, scenarios_list, config, failed_post_scenarios, wait_dur
...
+                    retry_container_killing(container_pod[0], container_pod[1], c_name)
+            else:
+                retry_container_killing(container_pod[0], container_pod[1], c_name)
+
+
+def retry_container_killing(podname, namespace, container_name):
+    i = 0
+    while i < 5:
+        logging.info("Killing container %s in pod %s (ns %s)" % (str(container_name), str(podname), str(namespace)))
+        response = kubecli.exec_cmd_in_pod("kill 1", podname, namespace, container_name)
```
Comment (chaitanyaenr) [inline kraken/pod_scenarios/setup.py:115]: Maybe expose the kill signal as part of the config? Thoughts?

Comment (mffiedler) [reply kraken/pod_scenarios/setup.py:115]: +1 for this.   Many OCP pods protect themselves by catching SIGINT.    Default to -2/SIGINT but allow arbitrary signals.   Or even random signals for more chaos fun!

Comment (paigerube14) [reply kraken/pod_scenarios/setup.py:115]: Sounds good! Will add this to the configs

File: kraken/pod_scenarios/setup.py (hunk 2 above)
Comment (paigerube14) [reply kraken/pod_scenarios/setup.py]: This was just a comment for my debugging, removed it

File: kraken/pod_scenarios/setup.py (hunk 3 above)
Comment (paigerube14) [reply kraken/pod_scenarios/setup.py]: PTAL at my updates

File: kraken/pod_scenarios/setup.py [hunk 4]
```diff
@@ -34,3 +37,85 @@ def run(kubeconfig_path, scenarios_list, config, failed_post_scenarios, wait_dur
...
+                'oc get pods %s -n %s -o jsonpath="{.spec.containers[*].name}"' % (pod[0], pod[1])
+            ).split(" ")
+            container_pod_list.append([pod[0], pod[1], container_names])
+        else:
+            container_names = runcommand.invoke(
+                'oc get pods %s -n %s -o jsonpath="{.spec.containers[*].name}"' % (pod, namespace)
+            ).split(" ")
+            container_pod_list.append([pod, namespace, container_names])
+
+    # jsonpath="{.items[*].spec.containers[*].name}"
```
Comment (chaitanyaenr) [inline kraken/pod_scenarios/setup.py:98]: should we remove this?

Comment (paigerube14) [reply kraken/pod_scenarios/setup.py]: removed

File: scenarios/container_etcd.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+scenarios:
```
Comment (chaitanyaenr) [inline scenarios/container_etcd.yml:1]: We might want to add a way to filter the number of pods to pick to kill the container. For example, killing the container in one of the etcd pods vs all 3 pods. Thoughts?

Comment (paigerube14) [reply scenarios/container_etcd.yml:1]: Updated, please take a look. Let me know if you can think of a better way to get the random pods and not keep killing the same one.

**Review by chaitanyaenr — APPROVED** [review]
LGTM

**Comment by mffiedler** [comment]
LGTM - I like using kill_action.  Ready to merge

**Pull request merged**